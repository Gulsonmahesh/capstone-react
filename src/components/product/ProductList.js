import React, { Component } from 'react'
import Product from './product';


class ProductList extends Component {
    render() {
        let selectedBrand = '';
        let selectedPrice = '';
        let productData = [];

        console.log(this.props.filterProduct);
        if (this.props.filterProduct !== null ){
            selectedBrand = (this.props.filterProduct.selectedBrand) ? this.props.filterProduct.selectedBrand.split(':')[1].trim() : '';
            selectedPrice = (this.props.filterProduct.selectedPrice) ? this.props.filterProduct.selectedPrice : '';
            if (selectedBrand === 'All') {
                productData = this.props.products;
            } else {
                productData = this.props.products.filter(product => product.modal === selectedBrand);
            }
        } else {
            productData = this.props.products;
        }
                
        return <Product productData={productData} />
    }
}

export default ProductList;
