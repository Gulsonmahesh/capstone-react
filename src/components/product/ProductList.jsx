import React, { Component } from 'react'
import Product from './product';


class ProductList extends Component {
    render() {
        let selectedBrand = '';
        // let selectedPrice = '';
        let productData = [];

        if (this.props.filterProduct !== null ){
            selectedBrand = (this.props.filterProduct.selectedBrand) ? (this.props.filterProduct.selectedBrand).trim() : '';
            // selectedPrice = (this.props.filterProduct.selectedPrice) ? this.props.filterProduct.selectedPrice : '';
            if (selectedBrand === 'All') {
                productData = this.props.products;
            } else {
                productData = this.props.products.filter(product => product.brand === selectedBrand);
            }
        } else {
            productData = this.props.products;
        }
                
        return <Product productData={productData} />
    }
}

export default ProductList;
