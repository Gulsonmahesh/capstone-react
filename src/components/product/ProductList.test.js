import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ProductList from './ProductList.jsx';

describe("Product list Snapshot ", () => {
    let props = {
        filterProduct : {
            selectedBrand : 'All'
        }
    }
    test("renders Correctly", () => {
        let enzymeWrapper = shallow(<ProductList {...props} />);
        console.log(enzymeWrapper)
        expect(toJson(enzymeWrapper)).toMatchSnapshot();
    })
});