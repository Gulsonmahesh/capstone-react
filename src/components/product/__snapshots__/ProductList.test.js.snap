// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list Snapshot  renders Correctly 1`] = `<Product />`;
